packages/x-tree-view/src/internals/plugins/useTreeViewExpansion/useTreeViewExpansion.ts

```diff
@@ -83,6 +83,17 @@ export const useTreeViewExpansion: TreeViewPlugin<UseTreeViewExpansionSignature>
     }
   };
 
+  const getExpansionTrigger = () => {
+    if (instance.isTreeViewEditable) {
+      return 'iconContainer';
+    }
+    if (params.expansionTrigger) {
```

Comment:
It's debatable, but I'd say that the prop should always have the priority.
I would switch the order of those 2 `if`

Reply:
If `params.expansionTrigger` has priority, and the user sets it to 'content' but does not change anything in the editing behavior, the parent items will behave really strange on double click. Is that ok? I would at least document it 

https://github.com/mui/mui-x/assets/72460825/7288be6b-4fba-43b6-9a59-174283329d2b

Reply:
But they would explicitly decide to have this awful behavior :laughing:  
You can always imagine weird edge cases where people want to have editing + trigger on the content because they have some custom editing API we did not think about.

If we have the internal value override `params.expansionTrigger`, then we should also document it btw.

Reply:
> But they would explicitly decide to have this awful behavior 

Also true. Okay, order switch it is 🆗

Reply:
Honestly it's a nitpick, I don't expect a lot of people to try this :laughing: